src/Domain/Insights/InsightFactory.php

```diff
@@ -80,29 +87,30 @@ public function makeFrom(string $errorClass, array $config): Sniff
     /**
      * Returns the Sniffs PHP CS classes from the given array of Metrics.
      *
-     * @param  array<string>  $insights
-     * @param  array<string, array>  $config
+     * @param array<string> $insights
+     * @param array<string, array> $config
+     * @param string $interface
      *
-     * @return array<\PHP_CodeSniffer\Sniffs\Sniff>
+     * @return array<SniffContract|FixerInterface>
      */
-    public function sniffsFrom(array $insights, array $config): array
+    public function insightsFrom(array $insights, array $config, string $interface): array
```

Comment:
Could you name this to be specific for phpcs instead? As the way it handles it is only for phpcs :)

Reply:
:thinking: I changed it to get Insights that implements any interface, so that not anymore specific to phpcs, as it could return Fixer from phpcsfixer. 

I think we could add simply support for more interface by just updating phpdoc, isn't it ?

Reply:
The issue is with the logic inside of it. For example in phpstan we do not define config by public properties on the class. The implementation here defines does that, and that way is how phpcs does it, in both sniffers and fixers.

Reply:
Okay I see what you mean :slightly_smiling_face: In fact, configuration for Fixer is not currently handled (public property doesn't work...) I have to make an improvment on that :wink:

Reply:
@olivernybroe Configuration is now handled in `configureInsight` method